test_communication/package.xml

```diff
@@ -11,21 +11,23 @@
   <maintainer email="dthomas@osrfoundation.org">Dirk Thomas</maintainer>
   <license>Apache License 2.0</license>
 
+  <build_depend>ament_cmake_auto</build_depend>
+  <build_depend>builtin_interfaces</build_depend>
+  <build_depend>rclcpp</build_depend>
+  <build_depend>rclpy</build_depend>
+  <build_depend>rcl</build_depend>
```

Comment:
I think that @wjwwood is right. You need the generators (rosidl_default_generators) to build the messages but not the client libraries. The client libraries can be used to interact with messages but are not required to generate/build them.